### PR TITLE
Use FileID instead of long to reference a file

### LINKED ISSUE #7098: Use FileID everywhere a File is referenced in PBJ/Proto definition or State
EntityNum and Long/int64 are insufficient to correctly reference a File because the bare number does not reference realm or shard, and this will prevent enabling sharding if not corrected.

### PR DESCRIPTION
**Pull request by kimbor**
Fixes #7098

### CODE REVIEW

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/codec/FileServiceStateTranslator.java
Comment (jsync-swirlds) [inline hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/codec/FileServiceStateTranslator.java:74]: Our style guide allows us to leave out the braces here (and the line above).
>### 4.1.1 Use of optional braces
>Single-line if statements are permitted even without braces.

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileUpdateHandler.java
Comment (jsync-swirlds) [inline hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileUpdateHandler.java:80]: No reason to have `requireNonNull(fileStore)`, `HandleContext.writableStore` does not return null (it's annotated `@NonNull`), so it can just be `fileStore`.

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/codec/FileServiceStateTranslator.java
Comment (kimbor) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/codec/FileServiceStateTranslator.java:74]: Tell that to Sonar. I added the braces because SonarLint was complaining.

Comment (jsync-swirlds) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/codec/FileServiceStateTranslator.java:74]: Sonar lint is wrong (I basically ignore sonar at this point, it's way off from what spotless wants, and sometimes making sonar not complain causes spotless to fail).
It's a minor item, and the braces are permitted, but it's also reasonable to leave them alone too.

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileUpdateHandler.java
Comment (kimbor) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileUpdateHandler.java:80]: That's not something I changed. I will remove the requireNonNull if I end up checking a different update for something else, but I don't think this is worth holding up the PR for.

Comment (jsync-swirlds) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileUpdateHandler.java:80]: Agreed, it's a minor item.

Comment (kimbor) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileUpdateHandler.java]: Fixed

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/schemas/GenesisSchema.java
Comment (rbair23) [inline hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/schemas/GenesisSchema.java:127]: We should file an issue to come back to these places and use the shard/realm defined in config for them

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java
Comment (rbair23) [inline hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java:213]: I guess this can be removed :)

File: settings.gradle.kts
Comment (rbair23) [inline settings.gradle.kts:106]: Make sure to coordinate with neeha and make sure the branch name you commit this PR with is correct after merging your changes in on the hedera-protobufs side

**Review by rbair23 — APPROVED** [review]

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/schemas/GenesisSchema.java
Comment (povolev15) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/schemas/GenesisSchema.java:127]: We need to create this issue for all services.

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/ReadableFileStoreImplTest.java
Comment (povolev15) [inline hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/ReadableFileStoreImplTest.java:45]: I think it is better also to compare file id to file id and not just the file num we are using it as an object the fileid

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java
Comment (povolev15) [inline hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java:223]: same here, compare also fileid

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java
Comment (povolev15) [inline hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java:254]: same here, compare also fileid

File: settings.gradle.kts
Comment (jsync-swirlds) [reply settings.gradle.kts:106]: This PR will merge with this branch to avoid the timing issue.
Then a subsequent PR will switch to the merged branch after the protobuf PR is merged.

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/schemas/GenesisSchema.java
Comment (kimbor) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/schemas/GenesisSchema.java]: I went ahead and implemented it for FileID. It was easy, quicker than filing an issue.

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java
Comment (kimbor) [reply hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java]: Fixed

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/ReadableFileStoreImplTest.java
Comment (kimbor) [reply hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/ReadableFileStoreImplTest.java]: Fixed

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java
Comment (kimbor) [reply hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java]: Fixed

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java
Comment (kimbor) [reply hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java]: Fixed

**Review by povolev15 — DISMISSED** [review]

File: hedera-node/hedera-file-service/src/main/java/com/hedera/node/app/service/file/FileMetadata.java
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-file-service/src/main/java/com/hedera/node/app/service/file/FileMetadata.java:37]: Do we need `FileMetadata` . Can we not use `File` object alone ? Not a blocker for this PR, just something to think about

File: settings.gradle.kts
Comment (Neeharika-Sompalli) [reply settings.gradle.kts:106]: If we are not merging `protobufs` PR first, then @kimbor  in order to avoid breaking develop, 
1. Merge this PR
2. Merge the protobufs PR, make sure this branch `07098-add-FileID` is not deleted. Since develop has this branch.
3. Then open PR to develop changing the branch name on develop to `add-pbj-types-to-state`

or as Jasper suggested in one call we keep the duplicate `fileNum` field in protobufs and remove it later.

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM !

File: hedera-node/hedera-file-service/src/main/java/com/hedera/node/app/service/file/FileMetadata.java
Comment (povolev15) [reply hedera-node/hedera-file-service/src/main/java/com/hedera/node/app/service/file/FileMetadata.java:37]: We can, I think but it is a similar path like in topic. let's create an issue to handle both.

**Review by rbair23 — DISMISSED** [review]

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java
Comment (qnswirlds) [inline hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java:115]: nit:
```suggestion
            for (final Key key : listKeys.keys()) {
```

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileTestUtils.java
Comment (qnswirlds) [inline hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileTestUtils.java:84]: nit:
```suggestion
        final FileID fileId = FileID.newBuilder().fileNum(2337).build();
```

**Review by qnswirlds — DISMISSED** [review]

**Review by JeffreyDallas — APPROVED** [review]

**Review by qnswirlds — APPROVED** [review]

**Review by povolev15 — APPROVED** [review]

**Review by qnswirlds — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]

**Review by rbair23 — APPROVED** [review]

**Pull request merged**
